predictsignauxfaibles/evaluate.py

```diff
@@ -128,24 +137,39 @@ def evaluate(
             If provided, the model will classify an entry X as positive if predict_proba(X)>=thresh.
             Otherwise, the model classifies X as positive if predict(X)=1, ie predict_proba(X)>=0.5
     """
+    aucpr = average_precision_score(
```

Comment:
I don't think `average_precision_score` computes the AUCPR  (see https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html#sklearn.metrics.average_precision_score) so you might want to rename this variable to avoid confusion :)

If that's what you really want, AUCPR can probably be computed using `sklearn.metrics.auc` (which gives you the area under any curve)

Reply:
Strictly speaking, it does compute the AUCPR, but in another way.
- **AUC with `sklearn.metrics.average_precision_score`**: Take the [formula of the average precision score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html#sklearn.metrics.average_precision_score):
$AP=\sum_{n}{(R_n - R_{n-1})*P_n}$
This is equivalent to computing the area under curve, without interpolation (ie, summing the areas under curve of each of the small blocks under the curve). This is an alternative way to compute the AUC.
- **AUC with `sklearn.metrics.auc`**: Function [`sklearn.metrics.auc`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.auc.html) also computes the area under curve, using a trapezoïdal rule. This can be slightly optimistic, as described in the doc of `average_precision_score`.

However, since `sklearn.metrics.auc` is the most common implementation, and in order not to confuse users, I'm substituting `sklearn.metrics.auc` here

Reply:
Actually let's stick to `average_precision_score` as its inputs are more intuitive, as they are the same as the other metrics:
- `average_precision_score` takes as inputs `y_true` and `y_score`, ie almost the same inputs as other metrics such as `balanced_accuracy_score`, `fbeta_score`, `precision_score`, etc
- `auc` takes as inputs a precision and a recall array, which means that both arrays would need to be recomputed within method `evaluate`, which would virtually be useless for the rest of the function

Reply:
Sounds good ! Thanks for clarifying, I learned something new today ! 🤓